mirage/happy_eyeballs_mirage.mli

```diff
@@ -1,17 +1,40 @@
-module Make (R : Mirage_random.S) (T : Mirage_time.S) (C : Mirage_clock.MCLOCK) (P : Mirage_clock.PCLOCK) (S : Tcpip.Stack.V4V6) : sig
-  module DNS : module type of Dns_client_mirage.Make(R)(T)(C)(P)(S)
```

Comment:
Somehow removing this functor application makes everything so complex -- now there's a "type dns" and "type flow" in the interface, and some strange "module Transport... with type io_addr = ... and type +'a io ...". Isn't there a way around it? Is the issue that OCaml's functors are generative (which can be avoided by https://discuss.ocaml.org/t/generative-functors/2945)?

Reply:
The main issue is to pass the `DNS.t` to the `connect` function. An other solution should be to create inside the `connect`  the expected `DNS.t` with a given `nameserver` argument (as `Dns_client_mirage`) but that mostly mean that: we don't create the `happy_eyeballs` device from a DNS device but from nothing. Are you ok with that?

Reply:
I see what you mean and need to think about it. At the moment it feels like the approach in this PR is fine, and apart from the type aliases (and implied complexity) provides us with a nice set of tradeoffs.